lumigator/python/mzai/backend/backend/tests/conftest.py

```diff
@@ -109,40 +116,47 @@ def db_session(db_engine: Engine):
+def fake_s3fs() -> S3FileSystem:
+    """Replace the filesystem registry for S3 with a MemoryFileSystem implementation."""
+    fsspec.register_implementation(
+        "s3", MemoryFileSystem, clobber=True, errtxt="Failed to register mock S3FS"
+    )
+    yield MemoryFileSystem()
+    print(f"final s3fs contents: {str(MemoryFileSystem.store)}")
 
 
-@pytest.fixture(scope="session", autouse=True)
-def setup_aws(localstack_container: LocalStackContainer):
-    """Setup env vars/AWS resources for use with the LocalStack container."""
-    # Initialize S3
-    s3 = localstack_container.get_client("s3")
-    try:
-        s3.create_bucket(
-            Bucket=settings.S3_BUCKET,
-            CreateBucketConfiguration={"LocationConstraint": localstack_container.region_name},
-        )
-    except ClientError as e:
-        if e.response["Error"]["Code"] == "BucketAlreadyOwnedByYou":
-            print("Bucket already created, continue")
-        else:
-            raise e
-    # add ENV vars for FSSPEC access to S3 (s3fs + HuggingFace datasets)
-    os.environ["FSSPEC_S3_KEY"] = "testcontainers-localstack"
-    os.environ["FSSPEC_S3_SECRET"] = "testcontainers-localstack"  # pragma: allowlist secret
-    os.environ["FSSPEC_S3_ENDPOINT_URL"] = localstack_container.get_url()
-    os.environ["AWS_ACCESS_KEY_ID"] = "testcontainers-localstack"
-    os.environ["AWS_SECRET_ACCESS_KEY"] = "testcontainers-localstack"  # pragma: allowlist secret
-    os.environ["AWS_ENDPOINT_URL"] = localstack_container.get_url()
+@pytest.fixture(scope="function")
+def fake_s3_client(fake_s3fs) -> S3Client:
+    """Provide a fake S3 client using MemoryFileSystem as underlying storage."""
+    os.environ["AWS_ACCESS_KEY_ID"] = "test"
+    # Please check https://github.com/localstack/localstack/issues/5894
+    # for info about the test region used
+    os.environ["AWS_DEFAULT_REGION"] = "us-east-1"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "test"  # pragma: allowlist secret
+    os.environ["AWS_ENDPOINT_URL"] = "http://example.com:4566"
+    return FakeS3Client(MemoryFileSystem.store)
 
 
 @pytest.fixture(scope="function")
-def s3_client(localstack_container: LocalStackContainer) -> S3Client:
-    return localstack_container.get_client("s3")
+def boto_s3_client() -> S3Client:
+    """Provide a real S3 client."""
+    os.environ["AWS_ACCESS_KEY_ID"] = "test"
+    # Please check https://github.com/localstack/localstack/issues/5894
+    # for info about the test region used
+    os.environ["AWS_DEFAULT_REGION"] = "us-east-1"
+    os.environ["AWS_SECRET_ACCESS_KEY"] = "test"  # pragma: allowlist secret
+    os.environ["AWS_ENDPOINT_URL"] = "http://localhost:4566"
+    return boto3.client("s3")
+
+
+@pytest.fixture(scope="function")
+def boto_s3fs() -> S3FileSystem:
+    """Provide a real s3fs client."""
+    s3fs = S3FileSystem()
+    mock_s3fs = Mock(wraps=s3fs)
+    yield mock_s3fs
+    print(f"intercepted s3fs calls: {str(mock_s3fs.mock_calls)}")
```

Comment:
remove print log (or alternatively switch to a logger)

Reply:
Moved to logger